include/deal.II/base/polynomial.h

```diff
@@ -142,6 +143,24 @@ namespace Polynomials
           const unsigned int n_derivatives,
           Number2 *          values) const;
 
+    /**
+     * Similar to the function above, but evaluate the polynomials on several
+     * positions at once, as described by the array argument @p points. This
+     * function is can be faster than the other function when the same
+     * polynomial should be evaluated on several positions at once, e.g., the
+     * x,y,z coordinates of a point for tensor-product polynomials.
+     *
+     * The template type `Number2` must implement arithmetic
+     * operations such as additions or multiplication with the type
+     * `number` of the polynomial, and must be convertible from
+     * `number` by `operator=`.
+     */
+    template <std::size_t dim, typename Number2>
```

Comment:
If I understand it correctly, `dim` is not actually the dimension but rather the vector size. For you application it is the same, but generally I find this notation here a bit confusing. What about `n_entries` or similar?

Reply:
That's a good idea, I changed to `n_entries`. (I was considering `size`, but that is so easy to confuse with the types so I decided against it.)